src/hal/utils/scope_vert.c

```diff
@@ -803,62 +806,67 @@ static void chan_sel_button(GtkWidget * widget, gpointer gdata)
-                                            GTK_DIALOG_MODAL,
-                                            GTK_MESSAGE_INFO,
-                                            GTK_BUTTONS_CLOSE,
-                                            _("Too many channels"));
-            gtk_message_dialog_format_secondary_text(
-                    GTK_MESSAGE_DIALOG(dialog),
-                    _("You cannot add another channel.\n\n"
-                    "Either turn off one or more channels, or shorten\n"
-                    "the record length to allow for more channels"));
-            gtk_dialog_run(GTK_DIALOG(dialog));
-            gtk_widget_destroy(dialog);
-	    return;
-	}
-	if (chan->name == NULL) {
-	    /* need to assign a source */
-	    if (dialog_select_source(chan_num) != TRUE) {
-		/* user failed to assign a source */
-		/* force the button to pop back out */
-		ignore_click = 1;
-		gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(widget), FALSE);
-		return;
-	    }
-	}
-	vert->chan_enabled[chan_num - 1] = 1;
+        }
+        count = 0;
+        for (n = 0; n < 16; n++) {
+            if (vert->chan_enabled[n]) {
+            count++;
+            }
+        }
+        if (count >= ctrl_shm->sample_len) {
+        /* max number of channels already enabled */
+        /* force the button to pop back out */
+            ignore_click = 1;
+            gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(widget), FALSE);
+                dialog = gtk_message_dialog_new(GTK_WINDOW(ctrl_usr->main_win),
+                                                GTK_DIALOG_MODAL,
+                                                GTK_MESSAGE_INFO,
+                                                GTK_BUTTONS_CLOSE,
+                                                _("Too many channels"));
+                gtk_message_dialog_format_secondary_text(
+                        GTK_MESSAGE_DIALOG(dialog),
+                        _("You cannot add another channel.\n\n"
+                        "Either turn off one or more channels, or shorten\n"
+                        "the record length to allow for more channels"));
+                gtk_dialog_run(GTK_DIALOG(dialog));
+                gtk_widget_destroy(dialog);
+            return;
+        }
+        if (chan->name == NULL) {
+        /* need to assign a source */
+            if (dialog_select_source(chan_num) != TRUE) {
+            /* user failed to assign a source */
+            /* force the button to pop back out */
+                ignore_click = 1;
+                gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(widget), FALSE);
+                return;
+            }
+        }
+        vert->chan_enabled[chan_num - 1] = 1;
+        /* make chan_num the selected channel */
+        vert->selected = chan_num;
+        channel_changed();
+    } else if (vert->selected == chan_num) {
+    /* a click on an already active channel turns it off */
+        set_channel_off(chan_num);
+        ignore_click = 0;
```

Comment:
The whitespace changes at the beginnings of these lines (and elsewhere in this branch) make the diff look larger than it really is.  It's not a big deal, but please try to avoid spurious whitespace changes like these.

The original code (before this branch) had a random mix of spaces and tabs, which is not ideal.  Our coding documentation, scattered and incomplete as it is, is silent on the spaces/tabs issue but requests to [avoid whitespace changes where possible](http://linuxcnc.org/docs/devel/html/code/contributing-to-linuxcnc.html#_follow_the_style_of_the_surrounding_code).

I fetched your branch and looked at it with `git log -p --ignore-space-change` to make it easier to see the meaningful changes.